frontend/lib/types/CompositeType.cpp

```diff
@@ -62,6 +64,16 @@ CompositeType::areSubsInstantiationOf(Context* context,
         // it was not an instantiation
         return false;
       }
+    } else {
+      // If the ID isn't found, then that means the generic component doesn't
+      // exist in the other type, which means this cannot be an instantiation
+      // of the other type.
+      //
+      // Currently this check assumes that 'this' and 'partial' share a root
+      // instantiation, so how could we reach this condition? One path here
+      // involves passing a tuple to a tuple formal with a fewer number of
+      // elements. For example, passing "(1, 2, 3)" to "(int, ?)".
+      return false;
```

Comment:
Looks like you have a test for this, so it works, but it's not clear to me why.

Reply:
The tuple size check is actually implemented in ``CanPassResult::canInstantiate``.

I had originally tried to address this in ``CompositeType::areSubsInstantiationOf``, but ``canInstantiate`` seemed more appropriate for focusing on tuples. I left the change to ``areSubsInstantiationOf`` because it seemed like the sort of thing that could come back to haunt us at some point.

I can remove the return from ``areSubsInstantiationOf`` change since it's not strictly necessary for this PR, and just leave the comment if you like.